tests/d3d12_dxvk_interop_device.c

```diff
@@ -0,0 +1,157 @@
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+#define VKD3D_DBG_CHANNEL VKD3D_DBG_CHANNEL_API
+#include "d3d12_crosstest.h"
+
+void test_vkd3d_dxvk_cmdbuf_interop(void)
+{
+    PFN_vkCmdUpdateBuffer pfn_vkCmdUpdateBuffer;
+    ID3D12DXVKInteropDevice1 *interop_device;
+    ID3D12GraphicsCommandList *command_list;
+    ID3D12CommandAllocator *allocator;
+    VkPhysicalDevice vkPhysicalDevice;
+    struct resource_readback rb;
+    struct test_context context;
+    ID3D12CommandQueue *queue;
+    VkCommandBuffer vkCmdBuf;
+    VkDeviceAddress offset;
+    ID3D12Resource *buffer;
+    ID3D12Device *device;
+    VkInstance vkInstance;
+    unsigned int value;
+    VkDevice vkDevice;
+    VkBuffer vkBuffer;
+    VkFormat format;
+    HRESULT hr;
+
+    static const unsigned int data_values[] = {0xdeadbeef, 0xf00baa, 0xdeadbeef, 0xf00baa};
+    static const uint32_t update_data[3] = { 0x1020304, 0xc0d0e0f, 0x5060708};
+
+    // Initialize pointer to GDPA function
+    if (!init_vulkan_loader())
+        return;
+
+    if (!init_test_context(&context, NULL))
+        return;
+
+    device = context.device;
+
+    if (FAILED(hr = ID3D12Device_QueryInterface(device,
+            &IID_ID3D12DXVKInteropDevice1, (void **)&interop_device)))
+    {
+        skip("ID3D12DXVKInteropDevice1 not implemented.\n");
+        destroy_test_context(&context);
+        return;
+    }
+
+    if (FAILED(hr = ID3D12DXVKInteropDevice1_GetVulkanHandles(interop_device,
+        &vkInstance,
+        &vkPhysicalDevice,
+        &vkDevice)))
+    {
+        ok(hr == S_OK, "ID3D12DXVKInteropDevice1_GetVulkanHandles failed %#x.\n", hr);
+        destroy_test_context(&context);
+        return;
+    }
+
+    pfn_vkCmdUpdateBuffer = (PFN_vkCmdUpdateBuffer)pfn_vkGetDeviceProcAddr(vkDevice, "vkCmdUpdateBuffer");
+    if (!pfn_vkCmdUpdateBuffer)
+    {
+        ok(pfn_vkCmdUpdateBuffer, "vkCmdUpdateBuffer not found.\n");
+        destroy_test_context(&context);
+        return;
+    }
+
+    if (FAILED(hr = ID3D12DXVKInteropDevice1_CreateInteropCommandAllocator(interop_device,
+        D3D12_COMMAND_LIST_TYPE_DIRECT,
+        VK_QUEUE_FAMILY_IGNORED,
+        &allocator)))
+    {
+        ok(hr == S_OK, "ID3D12DXVKInteropDevice1_CreateInteropCommandAllocator failed %#x.\n", hr);
+        destroy_test_context(&context);
+        return;
+    }
+
+    if (FAILED(hr = ID3D12Device_CreateCommandList(device, 0, D3D12_COMMAND_LIST_TYPE_DIRECT,
+            allocator, NULL, &IID_ID3D12GraphicsCommandList, (void **)&command_list)))
+    {
+        ok(hr == S_OK, "ID3D12Device_CreateCommandList failed %#x.\n", hr);
+        destroy_test_context(&context);
+        return;
+    }
+
+    if (FAILED(ID3D12DXVKInteropDevice1_BeginVkCommandBufferInterop(interop_device,
+        (ID3D12CommandList*)command_list,
+        &vkCmdBuf)))
+    {
+        ok(hr == S_OK, "ID3D12DXVKInteropDevice1_GetVulkanCmdHandles failed %#x.\n", hr);
```

Comment:
Shouldn't this now be something like the following?

```suggestion
    if (FAILED(hr = ID3D12DXVKInteropDevice1_BeginVkCommandBufferInterop(interop_device,
        (ID3D12CommandList*)command_list,
        &vkCmdBuf)))
    {
        ok(hr == S_OK, "ID3D12DXVKInteropDevice1_BeginVkCommandBufferInterop failed %#x.\n", hr);
```